mysql-test/t/mysqlslap.test

```diff
@@ -80,3 +80,13 @@ DROP DATABASE bug58090;
 
 --exec $MYSQL_SLAP --create-schema=test --init-command="CREATE TABLE t1(a INT)" --silent --concurrency=1 --iterations=1
 DROP TABLE t1;
+
+--echo #
+--echo # Bug MDEV-15789 (Upstream: #80329): MYSQLSLAP OPTIONS --AUTO-GENERATE-SQL-GUID-PRIMARY and --AUTO-GENERATE-SQL-SECONDARY-INDEXES DONT WORK
+--echo #
+
+--exec $MYSQL_SLAP --concurrency=1 --silent --iterations=1 --number-int-cols=2 --number-char-cols=2 --auto-generate-sql --auto-generate-sql-guid-primary --create-schema=slap
+DROP TABLE t1;
+--exec $MYSQL_SLAP --concurrency=1 --silent --iterations=1 --number-int-cols=2 --number-char-cols=2 --auto-generate-sql --auto-generate-sql-secondary-indexes=1 --create-schema=slap
+DROP TABLE t1;
```

Comment:
Why `DROP TABLE` here? FWICS there's no table `t1` created.

Reply:
Also MySQL does `--number-char-cols=3`, while in your patch it is `--number-char-cols=2`. Why was this change needed?

Reply:
I see t1 created even when the command fails (from both commands), hence the 2 DROP TABLEs.

As for --number-char-cols=2, I was just trying to make the table slightly smaller, while still demonstrating the error.  However, I'm happy to make them the same.